proposals/NNNN-non-escapable.md

```diff
@@ -0,0 +1,442 @@
+In particular, struct and enum initializers (which build a new value and return it to the caller) cannot be written without some mechanism similar to that outlined in our companion proposal.
+
+#### Globals and static variables cannot be nonescapable
+
+Nonescapable values must be constrained to some specific local execution context.
+This implies that they cannot be stored in global or static variables.
+
+#### Closures and nonescapable values
+
+Escaping closures cannot capture nonescapable values.
+Nonescaping closures can capture nonescapable values subject only to the usual exclusivity restrictions.
+
+Returning a nonescapable value from a closure requires explicit lifetime dependency annotations, as covered in the companion proposal.
+
+#### Nonescapable values and concurrency
+
+All of the requirements on use of nonescapable values as function arguments and return values also apply to async functions, including those invoked via `async let`.
+
+The closures used in `Task.init`, `Task.detached`, or `TaskGroup.addTask` are escaping closures and therefore cannot capture nonescapable values.
+
+#### Conditionally `Escapable` types
+
+You can define types whose escapability varies depending on their generic arguments.
+As with other conditional behaviors, this is expressed by using an extension to conditionally add a new capability to the type:
+
+```swift
+// Example: Conditionally Escapable generic type
+// By default, Box is itself nonescapable
+struct Box<T: ~Escapable>: ~Escapable {
+  var t: T
+}
+
+// Box gains the ability to escape whenever its
+// generic argument is Escapable
+extension Box: Escapable where T: Escapable { }
+```
+
+This can be used in conjunction with other suppressible protocols.
+For example, many general library types will need to be copyable and/or escapable following their contents.
+Here's a compact way to declare such a type:
+```swift
+struct Wrapper<T: ~Copyable & ~Escapable> { ... }
+extension Wrapper: Copyable where T: ~Escapable {}
+extension Wrapper: Escapable where T: ~Copyable {}
+```
+
+The above declarations all in a single source file will result in a type `Wrapper` that is `Escapable` exactly when `T` is `Escapable` and `Copyable` exactly when `T` is `Copyable`.
+To see why, first note that the explicit `extension Wrapper: Escapable` in the same source file implies that the original `struct Wrapper` must be `~Escapable` and similarly for `Copyable`, exactly as if the first line had been
+```swift
+struct Wrapper<T: ~Copyable & ~Escapable>: ~Copyable & ~Escapable {}
+```
+
+Now recall from SE-427 that suppressible protocols must be explicitly suppressed on type parameters in extensions.
+This means that
+```swift
+extension Wrapper: Copyable where T: ~Escapable {}
```

Comment:
Per SE-0427, this should be an error; the conditional conformance needs to explicitly declare whether it wants T to be copiable. (I.e., the only legal way to spell the conditional conformance is to use the long form below.)